packages/pigeon/lib/pigeon_lib.dart

```diff
@@ -1335,6 +1767,88 @@ class _RootBuilder extends dart_ast_visitor.RecursiveAstVisitor<Object?> {
             message: 'Expected a named type but found "$node".',
             lineNumber: _calculateLineNumber(source, node.offset)));
       }
+    } else if (_currentApi is AstProxyApi) {
```

Comment:
Can we extract all of this block to a helper?

Reply:
Done in https://github.com/flutter/packages/pull/5861